app/controllers/api/posts_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Api::PostsController < Api::ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments, :likes)
+    if @user
+      render json: @posts
+    else
+      render json: { status: 'Error', error: 'User Not Found' }
```

Comment:
- [ ] I noticed that the `load_and_authorize_resource` to load and authorize the `@posts` variable. However, you're not actually using the `@posts` variable in the rendering of the JSON response. Instead, you're using `@user.posts.` This may be a mistake, or it may be intentional, but it's worth double-checking to make sure that you're returning the correct data, kindly fix this :wink: